cpp/src/aztec/stdlib/merkle_tree/hash.test.cpp

```diff
@@ -11,17 +11,11 @@ TEST(stdlib_merkle_tree_hash, compress_native_vs_circuit)
     fr x = uint256_t(0x5ec473eb273a8011, 0x50160109385471ca, 0x2f3095267e02607d, 0x02586f4a39e69b86);
     Composer composer = Composer();
     witness_ct y = witness_ct(&composer, x);
-    field_ct z = plonk::stdlib::pedersen<Composer>::compress(y, y);
-    auto zz = crypto::pedersen::compress_native({ x, x });
-    EXPECT_EQ(z.get_value(), zz);
-}
+    field_ct z = plonk::stdlib::pedersen_hash<Composer>::hash_multiple({ y, y });
+    auto zz = crypto::pedersen_hash::hash_multiple({ x, x }); // uses fixed-base multiplication gate
+    if constexpr (Composer::type == ComposerType::PLOOKUP) {
```

Comment:
So we do not (and have not had in the past too) tests that check the native Pedersen hash. We always check if the native and circuit (i.e. `stdlib`) Pedersen hashes match. We do have tests that check the hash ladder (in `generators` module) but we don't need to test native implementation of Pedersen hash. The reason for that is there is no standard for Pedersen hash like we have for SHA and other hashes. In the other native hash implementations, we test against NIST-based test vectors. We don't have that for Pedersen hash. As long as native and stdlib implementations match, we are fine.

Reply:
BTW we still do test lookup based Pedersen hash because its design is much simpler. To write tests for fixed-base Pedersen hash is possible but not necessary in my opinion.